File: AppGenerateFiles/FrontModel/SecurityDataBaseModel.js
import { EntityClass } from "../../WDevCore/WModules/EntityClass.js";
import { WAjaxTools, BasicStates } from "../../WDevCore/WModules/WComponentsTools.js";
import { ModelProperty } from "../../WDevCore/WModules/CommonModel.js";
class Security_Permissions extends EntityClass {
   constructor(props) {
       super(props, 'EntitySecurity');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {Number}*/ Id_Permission;
   /**@type {String}*/ Descripcion;
   /**@type {String}*/ Estado;
   /**@type {Array<Security_Permissions_Roles>} OneToMany*/ Security_Permissions_Roles;
}
export { Security_Permissions }
class Security_Permissions_Roles extends EntityClass {
   constructor(props) {
       super(props, 'EntitySecurity');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {String}*/ Estado;
   /**@type {Security_Permissions} ManyToOne*/ Security_Permissions;
   /**@type {Security_Roles} ManyToOne*/ Security_Roles;
}
export { Security_Permissions_Roles }
class Security_Roles extends EntityClass {
   constructor(props) {
       super(props, 'EntitySecurity');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {Number}*/ Id_Role;
   /**@type {String}*/ Descripcion;
   /**@type {String}*/ Estado;
   /**@type {Array<Security_Permissions_Roles>} OneToMany*/ Security_Permissions_Roles;
   /**@type {Array<Security_Users_Roles>} OneToMany*/ Security_Users_Roles;
}
export { Security_Roles }
class Security_Users extends EntityClass {
   constructor(props) {
       super(props, 'EntitySecurity');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {Number}*/ Id_User;
   /**@type {String}*/ Nombres;
   /**@type {String}*/ Estado;
   /**@type {String}*/ Descripcion;
   /**@type {String}*/ Password;
   /**@type {String}*/ Mail;
   /**@type {String}*/ Token;
   /**@type {Date}*/ Token_Date;
   /**@type {Date}*/ Token_Expiration_Date;
   /**@type {Catalogo_Agentes} ManyToOne*/ Catalogo_Agentes;
   /**@type {Catalogo_Sucursales} ManyToOne*/ Catalogo_Sucursales;
   /**@type {Array<Transaction_Contratos_Inversionistas>} OneToMany*/ Transaction_Contratos_Inversionistas;
   /**@type {Array<Transaction_Contratos_old>} OneToMany*/ Transaction_Contratos_old;
   /**@type {Array<Transaction_Facturas_old>} OneToMany*/ Transaction_Facturas_old;
   /**@type {Array<Security_Users_Roles>} OneToMany*/ Security_Users_Roles;
}
export { Security_Users }
class Security_Users_Roles extends EntityClass {
   constructor(props) {
       super(props, 'EntitySecurity');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {String}*/ Estado;
   /**@type {Security_Roles} ManyToOne*/ Security_Roles;
   /**@type {Security_Users} ManyToOne*/ Security_Users;
}
export { Security_Users_Roles }

File: AppGenerateFiles/Model/SecurityDataBaseModel.cs
using CAPA_DATOS;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace DataBaseModel {
   public class Security_Permissions : EntityClass {
       [PrimaryKey(Identity = true)]
       public int? Id_Permission { get; set; }
       public string? Descripcion { get; set; }
       public string? Estado { get; set; }
       [OneToMany(TableName = "Security_Permissions_Roles", KeyColumn = "Id_Permission", ForeignKeyColumn = "Id_Permission")]
       public List<Security_Permissions_Roles>? Security_Permissions_Roles { get; set; }
   }
   public class Security_Permissions_Roles : EntityClass {
       [PrimaryKey(Identity = false)]
       public int? Id_Role { get; set; }
       [PrimaryKey(Identity = false)]
       public int? Id_Permission { get; set; }
       public string? Estado { get; set; }
       [ManyToOne(TableName = "Security_Permissions", KeyColumn = "Id_Permission", ForeignKeyColumn = "Id_Permission")]
       public Security_Permissions? Security_Permissions { get; set; }
       [ManyToOne(TableName = "Security_Roles", KeyColumn = "Id_Role", ForeignKeyColumn = "Id_Role")]
       public Security_Roles? Security_Roles { get; set; }
   }
   public class Security_Roles : EntityClass {
       [PrimaryKey(Identity = true)]
       public int? Id_Role { get; set; }
       public string? Descripcion { get; set; }
       public string? Estado { get; set; }
       [OneToMany(TableName = "Security_Permissions_Roles", KeyColumn = "Id_Role", ForeignKeyColumn = "Id_Role")]
       public List<Security_Permissions_Roles>? Security_Permissions_Roles { get; set; }
       [OneToMany(TableName = "Security_Users_Roles", KeyColumn = "Id_Role", ForeignKeyColumn = "Id_Role")]
       public List<Security_Users_Roles>? Security_Users_Roles { get; set; }
   }
   public class Security_Users : EntityClass {
       [PrimaryKey(Identity = true)]
       public int? Id_User { get; set; }
       public string? Nombres { get; set; }
       public string? Estado { get; set; }
       public string? Descripcion { get; set; }
       public string? Password { get; set; }
       public string? Mail { get; set; }
       public string? Token { get; set; }
       public DateTime? Token_Date { get; set; }
       public DateTime? Token_Expiration_Date { get; set; }
       public int? Id_Sucursal { get; set; }
       public int? id_agente { get; set; }
       [ManyToOne(TableName = "Catalogo_Agentes", KeyColumn = "id_agente", ForeignKeyColumn = "id_agente")]
       public Catalogo_Agentes? Catalogo_Agentes { get; set; }
       [ManyToOne(TableName = "Catalogo_Sucursales", KeyColumn = "Id_Sucursal", ForeignKeyColumn = "Id_Sucursal")]
       public Catalogo_Sucursales? Catalogo_Sucursales { get; set; }
       [OneToMany(TableName = "Transaction_Contratos_Inversionistas", KeyColumn = "Id_User", ForeignKeyColumn = "Id_User")]
       public List<Transaction_Contratos_Inversionistas>? Transaction_Contratos_Inversionistas { get; set; }
       [OneToMany(TableName = "Transaction_Contratos_old", KeyColumn = "Id_User", ForeignKeyColumn = "Id_User")]
       public List<Transaction_Contratos_old>? Transaction_Contratos_old { get; set; }
       [OneToMany(TableName = "Transaction_Facturas_old", KeyColumn = "Id_User", ForeignKeyColumn = "Id_User")]
       public List<Transaction_Facturas_old>? Transaction_Facturas_old { get; set; }
       [OneToMany(TableName = "Security_Users_Roles", KeyColumn = "Id_User", ForeignKeyColumn = "Id_User")]
       public List<Security_Users_Roles>? Security_Users_Roles { get; set; }
   }
   public class Security_Users_Roles : EntityClass {
       [PrimaryKey(Identity = false)]
       public int? Id_Role { get; set; }
       [PrimaryKey(Identity = false)]
       public int? Id_User { get; set; }
       public string? Estado { get; set; }
       [ManyToOne(TableName = "Security_Roles", KeyColumn = "Id_Role", ForeignKeyColumn = "Id_Role")]
       public Security_Roles? Security_Roles { get; set; }
       [ManyToOne(TableName = "Security_Users", KeyColumn = "Id_User", ForeignKeyColumn = "Id_User")]
       public Security_Users? Security_Users { get; set; }
   }
}
